### ISSUE TITLE
User preference to automatically open the content warning (self-label) dialog when attaching media

### ISSUE THREAD

**Issue by nasso** [bluesky-social/social-app#6815]
### Describe the Feature

TL;DR: Add a user preference checkbox that makes the content warning (self-label) modal dialog open automatically when media is added in the composer.

## Motivation

Content creators and artists who often post adult or graphic content complain that they tend to forget to set labels before publishing their work. Because self-labels can't be applied after the post was published, the only way is to wait on moderation services to catch that mistake, adding extra load on the shoulders of the moderation team.

After discussing with the relevant users, they've expressed their desire to have a more aggressive reminder that could be enabled in the user preferences.

## Proposed solution

I think the best way to address this is to make self-labeling part of the user flow when composing a post with media. I believe the best way to do that is to open the label selection dialog as soon as media is attached. Because this is opt-in, it shouldn't be annoying, especially for users who post a lot of media that needs self-labelling.

### Attachments

https://github.com/user-attachments/assets/f1250f41-1cd3-427f-82c2-279543d7ca8e

### Describe Alternatives

1. Show the dialog when clicking "Post". The dialog would need a "Cancel" button though, to go back to the composer. It's unclear whether "Cancel" would keep the label selection though.
2. Show a confirmation dialog when hitting "Post" if the post has media attached without any label.
3. Show a warning in the composer when the post has media attached without a label. This would be similar to the existing warning for missing ALT text, but without enforcing labels to be added. This is probably less effective.
4. Same as (3) but add a "safe" option in the label selection dialog form, and enforce it to be explicitly selected. This is Confusing™ and Weird™.

### Additional Context

Related: https://github.com/bluesky-social/social-app/issues/5948, https://github.com/bluesky-social/social-app/issues/5628

**Comment by foxyoreos** [comment]
Copying over from #5628 

Just being able to set a default that I can override would be a large usability improvement, and from my experience on other social networks, NSFW artists would use the feature.

If I could set a default label, it would make posting on this network a lot less stressful, because if I post something SFW and accidentally forget to remove the label - oops, but no big deal. If I post something NSFW and forget to add the label, I have to wait for auto-moderation to catch it (if it even does), and in the meantime all of my followers get spammed with unlabeled porn. That's assuming I even notice the mistake.

In the meantime, because tensions on Bluesky are relatively high right now about NSFW art and labeling, I'm much more likely to get added to a bunch of "doesn't label their posts" blocklists if I ever make that mistake. >w<

Basically, this is a foot-gun waiting to happen. A default label that I can override or a more aggressive upload flow that I can enable doesn't fix all of the problems, but it fixes a lot of them without adding a lot of complexity. Nothing needs to be added to the protocol, it can be a purely clientside feature (although it would be nice to have the setting sync). It would reduce my stress and make the network safer.
